### PR TITLE
Collection, List, DDList, SelectBox, TagBox: replace all ds requests with DataController requests

### PR DESCRIPTION
**Pull request by Zedwag**

### CODE REVIEW

File: js/core/data_controller.js [hunk 1]
```diff
@@ -0,0 +1,173 @@
...
+    loadOptions() {
+        return this._dataSource?.loadOptions();
+    }
+
+    userData() {
+        return this._dataSource?._userData;
+    }
+
+    cancel(operationId) {
+        this._dataSource?.cancel(operationId);
```
Comment (ksercs) [inline js/core/data_controller.js]: let's use [null object pattern](https://sourcemaking.com/design_patterns/null_object) to get rid of all this optional chainings and conditions. You can find an example here: https://github.com/DevExpress/DevExtreme/blob/22_2/js/core/resize_observer.js

Comment (Zedwag) [reply js/core/data_controller.js]: done

File: js/ui/collection/ui.collection_widget.edit.js [hunk 2]
```diff
@@ -215,37 +214,34 @@ const CollectionWidget = BaseCollectionWidget.inherit({
             filter: that._getCombinedFilter.bind(that),
             totalCount: function() {
                 const items = that.option('items');
-                const dataSource = that._dataSource;
-                return dataSource && dataSource.totalCount() >= 0
-                    ? dataSource.totalCount()
+                const dataController = that._dataController;
+                return dataController.totalCount() >= 0
```
Comment (ksercs) [inline js/ui/collection/ui.collection_widget.edit.js]: let's remember the result of `dataController.totalCount()` to not call function two times

File: js/ui/editor/ui.data_expression.js [hunk 3]
```diff
@@ -41,6 +42,7 @@ const DataExpressionMixin = extend({}, DataHelperMixin, {
                 store: new ArrayStore(this.option('items')),
                 pageSize: 0
             });
+            this._dataController = new DataController(this._dataSource);
```
Comment (ksercs) [inline js/ui/editor/ui.data_expression.js]: why do we create dataSource here manually, not use `initDataController` function?

Comment (Zedwag) [reply js/ui/editor/ui.data_expression.js]: done

File: js/ui/collection/ui.collection_widget.edit.js (hunk 2 above)
Comment (Zedwag) [reply js/ui/collection/ui.collection_widget.edit.js]: done

**Review by ksercs — DISMISSED** [review]

**Review by EugeniyKiyashko — APPROVED** [review]

**Review by ksercs — APPROVED** [review]

**Pull request merged**